app/models/pet_application.rb

```diff
@@ -0,0 +1,16 @@
+class PetApplication < ApplicationRecord
+  validates :name, presence: true
+  validates :street, presence: true
+  validates :city, presence: true
+  validates :state, presence: true
+  validates :zip, presence: true
+  validates :description, presence: true
+
+  has_many :pet_application_pets
+  has_many :pets, through: :pet_application_pets
+
+  def status_change(status)
+    self.status = status
+    save
```

Comment:
If you do the status change in the controller before you cal `@application.save` then you would not need to call `save` here.